views/pages/ConfigurationsPage/index.jsx

```diff
@@ -0,0 +1,164 @@
+import { Add } from '@mui/icons-material'
+import { useOutletContext, Link } from 'react-router-dom'
+
+import ConfigurationsTable from '../../tables/ConfigurationsTable'
+import PageHeader from '../../components/PageHeader'
+import { UsersModel } from '../../models'
+import api from '../../api'
+import ShareConfigurationForm from '../../forms/ShareConfigurationForm'
+import { routes } from '../../routes/routes'
+
+const ConfigurationsPage = () => {
+  const {
+    configurations,
+    navigate,
+    setNotification,
+    setConfigurations,
+    setUser,
+    user,
+    users,
+  } = useOutletContext()
+  const { uid, preferences } = user
+  const [open, setOpen] = useState(false)
+  const [options, setOptions] = useState([])
+  const [formValues, setFormValues] = useState([])
+  const [currentConfigId, setCurrentConfigId] = useState('')
+
+  const handleNotification = (message) =>
+    setNotification({ open: true, message })
+  const onSubmit = async (data) => {
+    try {
+      const configAttributes = {
+        readers: UsersModel.createReadersList(data.readers),
+      }
+      await api.userConfigurations.update(
+        uid,
+        currentConfigId,
+        configAttributes
+      )
+
+      await loadAllConfigurations(uid)
+
+      closeForm()
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const loadAllConfigurations = async (userId) => {
+    try {
+      const configurations = await api.userConfigurations.all(userId)
+
+      setConfigurations(configurations)
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const closeForm = () => {
+    setCurrentConfigId('')
+    setOpen(false)
+    setOptions([])
+    reset()
+  }
+  const handleFormModal = async (config) => {
+    const { owner, readers, _id } = config
+    const options = UsersModel.createUserFriendList(users, owner)
+    const values = UsersModel.createOptionsFromReaders(readers, owner)
+
+    setCurrentConfigId(_id)
+    setFormValues(values)
+    setOptions(options)
+    setOpen(true)
+  }
+  const handleDefaultConfiguration = async (configurationId) => {
+    try {
+      const userAttributes = {
+        ...user,
+        preferences: {
+          ...preferences,
+          config: preferences.config === configurationId ? '' : configurationId,
+        },
+      }
+      const updatedUser = await api.users.update(uid, userAttributes)
+
+      setUser(updatedUser)
+      handleNotification('Default Configuration Updated Successfully ')
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const handleEditConfig = (configurationId) =>
+    navigate(routes.editConfiguration(configurationId))
+  const handleDeleteConfig = async (configurationId) => {
+    try {
+      await api.userConfigurations.destroy(uid, configurationId)
+
+      await loadAllConfigurations(uid)
+
+      if (preferences.config === configurationId)
+        handleDefaultConfiguration(configurationId)
```

Comment:
this method looks like it is going to either set or remove the default configuration, but we just deleted it. Should we be making sure we remove it as the default option instead of toggling it?

Reply:
yes the handleDefaultConfiguration takes care of removing the configuration

Reply:
only adds it if the configuration id is different than what's currently in the users preferences. In the case above we are checking if the id's match and if they do then we set it to empty, if they are different then we are adding it

Reply:
[3f060e8](https://github.com/AMP-SCZ/dpdash/pull/487/commits/3f060e8279d72eb848b54d8b7eb594892d7449a8)